Update

Dreeam-qwq opened the pull request:
Changes:

* Update to Gradle 8.6
* Convert Gradle to kotlin DSL
* Improve multi version compatibility for Material
* Cleanup code

Dreeam-qwq committed:
Fix 1.12.2 compatibility

Dreeam-qwq committed:
Update

* Update to Gradle 8.6
* Convert Gradle to kotlin DSL
* Improve multi version compatibility for Material
* Cleanup code

Dreeam-qwq committed:
Fixes

Ivan8or commented on src/main/java/online/umbcraft/libraries/utils/MaterialUtil.java:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    public static boolean isAnvil(Material m) {
+        return m == ((GoldenDupes.getInstance().serverVersion() <= 12) ? Material.valueOf("ANVIL") : Material.DAMAGED_ANVIL);
```
Is this enough to tell if the anvil is in its damaged state when on version<=1.12? To me it seems like this just checks if the block is any anvil, not specifically a damaged anvil. I believe you have to check the block data value as well to correctly support older versions.
(for example, in 1.8 jungle planks are `minecraft:planks:3`, similarly damaged anvils were `minecraft:anvil:2`)

Dreeam-qwq replied:
Yes, this only checks the Material is anvil, and need to add getData check to see whether a damaged_anvil. I didn't consider this before.

Ivan8or reviewed (changes requested):
Looks fine for the most part, I'm pretty neutral about the switch to kotlin gradle but if it's preferred by people in general then so be it.

Dreeam-qwq committed:
Improve 1.12.2 compatibility

In 1.12.2 (Before 1.13 Flattening), it's need to use Block.getData() to get data value of a block. And the data value of Anvil in 1.12.2 is:
anvil -> 2
chipped_anvil -> 6
damaged_anvil -> 10

Ivan8or commented on src/main/java/online/umbcraft/libraries/utils/MaterialUtil.java:
```diff
@@ -0,0 +1,27 @@
+
+    public static boolean isTotem(Material m) {
+        if (GoldenDupes.getInstance().serverVersion() <= 12) {
+            return m == Material.valueOf("TOTEM");
```
I believe this code may break when running on versions under 1.11, since totems were added in 1.11. Calling valueOf on an enum which doesn't exist will throw an IllegalArgumentException - this case can probably be solved either with another version check which simply returns false if the version is below 11, or (less ideally) by catching the exception as the previous code did

Dreeam-qwq committed:
return false for isTotem & isShulkerBox if under 1.11

Dreeam-qwq committed:
Cache it
